File: Encrypt.java
import java.util.*;
class Encrypt{

    Scanner rc =new Scanner(System.in);
    int a,b;
int gcd(int a,int b)
{
    int rem=a%b;
    // Finding GCD using Euclidean Alogrithm
    while(rem!=0)
    {
        a=b;
        b=rem;
        rem=a%b;
    }
    return b;
}

int inversemod(int num,int mod)
{
    for (int i = 1; i < mod; i++)
        if (((num % mod) * (i % mod)) % mod == 1)
            return i;
    return 0;
}

int mod_arithmetic(int num,int mod)
{
    if (num>=0)
        return num%mod;
    while (num < 0)
        num += mod;
    return num % mod;
}

void affineD()
{
    Scanner r=new Scanner(System.in);
    System.out.print("Enter a string: ");
    String s=r.nextLine().toUpperCase();
    String temp="";
    for (int i=0;i<s.length();i++)
    {
        int c=((int)s.charAt(i))-65;
        int temp1=inversemod(a, 26)*(c-b);
        int p=mod_arithmetic(temp1,26)+65;
        temp=temp+((char)p);
    }
    System.out.println("Decrypted Text: ");
    System.out.println(temp);
}

void affineE()
{
    Scanner r=new Scanner(System.in);
    System.out.print("Enter a string: ");
    String s=r.nextLine().toUpperCase();
    String temp="";
    for (int i=0;i<s.length();i++)
    {
        int p=((int)s.charAt(i))-65;
        int temp1=(a*p)+b;
        int c=mod_arithmetic(temp1,26)+65;
        temp=temp+((char)c);
    }
    System.out.println("Encrypted Text: ");
    System.out.println(temp);
}

void print()
{
    // System.out.println("GCD = "+gcd(252,198));
    System.out.println("Enter \'A\':");
    a=rc.nextInt();
    while(gcd(a,26)!=1)
    {
        System.out.println("Enter valid \'A\':");
        a=rc.nextInt();
        if (gcd(a, 26)==1)
        {
            break;
        }
    }
    System.out.println("Enter \'B\':");
    b=rc.nextInt();
    System.out.print("\tMenu\n1. Encrypt\n2. Decrypt\n3. Exit\nEnter your Choice: ");
    int choice=rc.nextInt();
    System.out.println();
    switch(choice)
    {
        case 1: affineE();
                break;
        case 2: affineD();
                break;
        case 3: System.exit(0);
                break;
        default:System.out.println("You have enter worng choice!");
    }
}

public static void main(String[] args)
{
    Encrypt e = new Encrypt();
    e.print();
}
}

File: Encrypt.py

a=0
b=0
def gcd(a,b):
    rem=a%b
    while rem!=0:
        a=b
        b=rem
        rem=a%b
    return b

def inversemod(num,mod):
    for i in range(1,mod):
        if (((num % mod) * (i % mod)) % mod == 1):
            return i
        
def mod_arithmetic(num,mod):
    if num>=0:
        return num%mod
    while num<0:
        num+=mod
    return num % mod

def affineD():
    global a,b
    s=input("Enter a string:").upper()
    temp=""
    for i in s:
        c=ord(i)-65
        temp1=inversemod(a,26)*(c-b)
        p=int(mod_arithmetic(temp1,26))+65
        temp=temp+chr(p)
    print(f"Decrypted Text: {temp}")

def affineE():
    global a,b
    s=input("Enter a string:").upper()
    temp=""
    for i in s:
        p=ord(i)-65
        temp1=(a*p)+b
        c=mod_arithmetic(temp1,26)+65
        temp=temp+chr(c)
    print(f"Encrypted Text: {temp}")

def main():
    global a,b
    a=int(input("Enter \'A\': "))
    while gcd(a,26)!=1:
        a=int(input("Enter valid \'A\'"))
        if gcd(a,26)==1:
            break
    b=int(input("Enter \'B\': "))
    choice=int(input("\tMenu\n1. Encrypt\n2. Decrypt\n3. Exit\nEnter your Choice: "))
    if choice==1:
        affineE()
    elif choice==2:
        affineD()
    
if __name__=='__main__':
    main()
